packages/teams-js/src/public/visualMedia.ts

```diff
@@ -0,0 +1,301 @@
+     *
+     * @beta
+     */
+    export async function captureImages(cameraImageInputs: CameraImageProperties): Promise<VisualMediaFile[]> {
+      ensureInitialized(runtime, FrameContexts.content, FrameContexts.task);
+      ensureSupported();
+      ensureImageInputValided(cameraImageInputs);
+      // waiting the response from host apps
+      const files = await sendAndHandleSdkError<VisualMediaFile[]>(
+        'visualMedia.image.captureImages',
+        cameraImageInputs,
+      );
+      ensureResponseValided(cameraImageInputs.maxVisualMediaCount, files);
+      return files;
+    }
+
+    /**
+     * @hidden
+     * Upload the existing image(s) from the gallery.
+     * @param galleryImageInputs - The input params to customize the image(s) to be captured
+     * @returns Promise that will resolve with {@link VisualMediaFile[]} object or reject with an error.
+     * @throws INVALID_ARGUMENTS Error if imageInputs is null or imageInputs.maxVisualMediaCount is greater than maxVisualMediaSelectionLimit or lesser than 1.
+     *
+     * @beta
+     */
+    export async function retrieveImages(galleryImageInputs: GalleryImageProperties): Promise<VisualMediaFile[]> {
+      ensureInitialized(runtime, FrameContexts.content, FrameContexts.task);
+      ensureSupported();
+      ensureImageInputValided(galleryImageInputs);
+      // waiting the response from host apps
+      const files = await sendAndHandleSdkError<VisualMediaFile[]>(
+        'visualMedia.image.retrieveImages',
+        galleryImageInputs,
+      );
+      ensureResponseValided(galleryImageInputs.maxVisualMediaCount, files);
+      return files;
+    }
+
+    /**
+     * @hidden
+     * Checks if visualMedia.image capability is supported by the host
+     * @returns boolean to represent whether visualMedia.image is supported
+     * @throws Error if {@linkcode app.initialize} has not successfully completed
+     *
+     * @beta
+     */
+    export function isSupported(): boolean {
+      return ensureInitialized(runtime) &&
+        runtime.supports.visualMedia &&
+        runtime.supports.visualMedia.image &&
+        runtime.supports.permissions
+        ? true
+        : false;
+    }
+
+    /**
+     * @hidden
+     * Ensure visualMedia.image capability is supported by the host
+     * @throws errorNotSupportedOnPlatform error if isSupported() fails.
+     *
+     * @beta
+     */
+    function ensureSupported(): void {
+      if (!isSupported()) {
+        throw errorNotSupportedOnPlatform;
+      }
+    }
+    /**
+     * @hidden
+     * @param imageInput the input can be either CameraImageProperties or GalleryImageProperties
+     * @param source the expected Source
+     * @throws error if the input check fails.
+     * @beta
+     */
+    function ensureImageInputValided(imageInput: CameraImageProperties | GalleryImageProperties): void {
```

Comment:
```suggestion
    function ensureImageInputValid(imageInput: CameraImageProperties | GalleryImageProperties): void {
```

You could also go with `validateImageInput` if you prefer that